doc/topics/packaging/testing.rst

```diff
@@ -0,0 +1,117 @@
+
+        tools ts setup --platform {linux|darwin|windows} --slug <operating-system-slug> --pr <pr-number> --pkg
+
+    To see the possible options for each argument, and other ways to utilize this command, use the following.
+
+    .. code-block:: bash
+
+        tools ts setup -h
+
+
+Downloading Individually
+------------------------
+If the ``tools ts setup`` command doesn't quite work, the artifacts can be downloaded, unzipped, and placed in the correct locations manually.
+Most often, you will be testing packages built on a CI/CD run for a given PR. This guide will explain how to setup for running the package tests using those artifacts.
+An analogous process can be performed for artifacts from nightly builds.
+
+#. Download the artifacts
+
+    Under the summary page for the most recent actions run for that PR, there is a list of available artifacts from that run that can be downloaded.
+    Download the package artifacts by finding ``salt-<major>.<minor>+<number>.<sha>-<arch>-<pkg-type>``.  For example, the amd64 deb packages
+    might look like ``salt-3006.2+123.01234567890-x86_64-deb``.
+
+    The onedir artifact will look like ``salt-<major>.<minor>+<number>.<sha>-onedir-<platform>-<arch>.tar.xz``.
+    For instance, the macos x86_64 onedir may have the name ``salt-3006.2+123.01234567890-onedir-darwin-x86_64.tar.xz``.
+
+    .. note::
+
+        Windows onedir artifacts have ``.zip`` extensions instead of ``tar.xz``
+
+    While it is optional, it is recommended to download the ``nox`` session artifact as well.  This will have the form of ``nox-<os-name>-test-pkgs-onedir-<arch>``.
+    The amd64 Ubuntu 20.04 nox artifact may look like ``nox-ubuntu-20.04-test-pkgs-onedir-x86_64``.
+
+#. Placing the artifacts in the correct location
+
+    The packages should be unzipped and placed under ``<repo-root>/pkg/artifacts/``.
+
+    The onedir artifact must be unzipped and untarred (or unzipped again, for windows onedirs) under ``<repo-root>/artifacts/``.
+    There should then be a ``<repo-root>/artifacts/salt`` directory that contains the uncompressed onedir.
+
+    Lastly, the ``nox`` artifact should be fully uncompressed and placed under ``<repo-root>/.nox/``.
+
+Running the Tests
+=================
+Once all the artifacts are in the correct locations, the test suite can now be run.
+
+.. note::
+
+    All nox commands should be run at the root of the salt repo and as the root user.
+
+#. Install ``nox``
+
+    .. code-block:: bash
+
+        pip install nox
+
+#. Run the install tests
+
+    .. code-block:: bash
+
+        nox -e test-pkgs-onedir -- install
+
+#. Run the upgrade or downgrade tests
+
+    .. code-block:: bash
+
+        nox -e test-pkgs-onedir -- upgrade --prev-version <previous-version>
+
+    The downgrade tests can be run in the same way, replacing ``upgrade`` with ``downgrade``.
+
+    .. note::
+
+        If the previous version being tested is before 3006.0 and there are classic packages available for your system,
```

Comment:
Active voice: "If you are testing a version of Salt prior to 3006.0 and classic packages are available for your system..."